src/main/java/bg/libapp/libraryapp/service/BookService.java

```diff
@@ -0,0 +1,82 @@
+package bg.libapp.libraryapp.service;
+
+import bg.libapp.libraryapp.exceptions.AuthorNotFoundException;
+import bg.libapp.libraryapp.exceptions.BookAlreadyAddedException;
+import bg.libapp.libraryapp.exceptions.BookNotFoundException;
+import bg.libapp.libraryapp.exceptions.GenreNotFoundException;
+import bg.libapp.libraryapp.model.dto.book.BookAddDTO;
+import bg.libapp.libraryapp.model.dto.book.BookViewDTO;
+import bg.libapp.libraryapp.model.entity.Author;
+import bg.libapp.libraryapp.model.entity.Book;
+import bg.libapp.libraryapp.model.entity.Genre;
+import bg.libapp.libraryapp.model.mappers.BookMapper;
+import bg.libapp.libraryapp.repository.AuthorRepository;
+import bg.libapp.libraryapp.repository.BookRepository;
+import bg.libapp.libraryapp.repository.GenreRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+public class BookService {
```

Comment:
Research about @Transactional annotation. Think of where should it be used.

Reply:
Anywhere where there is some updates of database?